clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -107,7 +108,10 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, HostResolver hostResolver) throws UnknownHostException {
         InetAddress[] addresses = hostResolver.resolve(host);
-        return filterPreferredAddresses(addresses);
+        List<InetAddress> result = filterPreferredAddresses(addresses);
+        if (log.isDebugEnabled())
+            log.debug("Resolved host {} as {}", host, result.stream().map(i -> i.getHostAddress()).collect(Collectors.toList()));
```

Comment:
If you're going to convert to a String anyway `Collections.joining()` would be better.

Reply:
thanks, going for `Collectors.joining(",")` I suppose spaces are not needed